go-controller/pkg/ovn/zone_interconnect/interconnect.go

```diff
@@ -0,0 +1,524 @@
+	}
+
+	if err := libovsdbops.DeleteLogicalSwitchPorts(zic.nbClient, logicalSwitch, logicalSwitchPort); err != nil {
+		return fmt.Errorf("failed to delete logical switch port %s from transit switch %s for the node %s, error: %v", logicalSwitchPort.Name, zic.networkTransitSwitchName, node.Name, err)
+	}
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) setRemotePortBindingChassis(nodeName, portName, chassisId string) error {
+	remotePort := sbdb.PortBinding{
+		LogicalPort: portName,
+	}
+	chassis := sbdb.Chassis{
+		Hostname: nodeName,
+		Name:     chassisId,
+	}
+
+	if err := libovsdbops.UpdatePortBindingSetChassis(zic.sbClient, &remotePort, &chassis); err != nil {
+		return fmt.Errorf("failed to update chassis %s for remote port %s, error: %v", nodeName, portName, err)
+	}
+
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node, nodeID int, nodeTransitSwitchPortIPs []*net.IPNet) error {
+	addRoute := func(subnet *net.IPNet, tsIp string) error {
+		logicalRouterStaticRoute := nbdb.LogicalRouterStaticRoute{
+			ExternalIDs: map[string]string{
+				"ic-node": node.Name,
+			},
+			Nexthop:  tsIp,
+			IPPrefix: subnet.String(),
+		}
+		p := func(lrsr *nbdb.LogicalRouterStaticRoute) bool {
+			return lrsr.IPPrefix == subnet.String() &&
+				lrsr.Nexthop == tsIp &&
+				lrsr.ExternalIDs["ic-node"] == node.Name
+		}
+		//FIXME: All routes should be added in a single transaction instead.
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(zic.nbClient, zic.networkClusterRouterName, &logicalRouterStaticRoute, p); err != nil {
+			return fmt.Errorf("failed to create static route: %v", err)
+		}
+		return nil
+	}
+
+	nodeSubnets, err := util.ParseNodeHostSubnetAnnotation(node, zic.GetNetworkName())
+	if err != nil {
+		return fmt.Errorf("failed to parse node %s subnets annotation %v", node.Name, err)
+	}
+
+	//FIXME: No consistency on transaction failure.
+	for _, subnet := range nodeSubnets {
+		for _, tsIP := range nodeTransitSwitchPortIPs {
+			if utilnet.IPFamilyOfCIDR(subnet) != utilnet.IPFamilyOfCIDR(tsIP) {
+				continue
+			}
+			if err := addRoute(subnet, tsIP.IP.String()); err != nil {
+				return fmt.Errorf("unable to create static routes: %v", err)
+			}
+
+		}
+	}
+
+	if zic.IsSecondary() {
+		// Secondary network cluster router doesn't connect to a join switch
+		// or to a Gateway router.
+		return nil
+	}
+
+	nodeGWRPIPs, err := util.ParseNodeGatewayRouterLRPAddrs(node)
+	if err != nil {
+		return fmt.Errorf("failed to parse node %s Gateway router LRP Addrs annotation %v", node.Name, err)
+	}
+
+	for _, gwRPAddr := range nodeGWRPIPs {
+		for _, tsIP := range nodeTransitSwitchPortIPs {
+			if utilnet.IPFamilyOfCIDR(gwRPAddr) != utilnet.IPFamilyOfCIDR(tsIP) {
+				continue
+			}
+			hostRoute := &net.IPNet{IP: gwRPAddr.IP}
+			if utilnet.IsIPv4CIDR(gwRPAddr) {
+				hostRoute.Mask = net.CIDRMask(32, 32)
```

Comment:
maybe write a util function similar to `util.GetIPFullMask` for this? too much low level logic in an already heavy function.